### PR TITLE
[oesgx] epc subleaf memory calculation

### PR DESCRIPTION
**Pull request by RRathna**
correcting EPC memory calculation by iterating over ecx to find out if _CPUID() passes, and subleaf exists.

### CODE REVIEW

File: tools/oesgx/oesgx.c [hunk 1]
```diff
@@ -162,15 +163,24 @@ int main(int argc, const char* argv[])
...
+
+        while(!result && HAVE_EPC_SUBLEAF(regs))
+        {
+            epc_size +=
+                ((regs.ecx & 0x0fffff000) |
+                 ((uint64_t)(regs.edx & 0x0fffff) << 32));
+
+            ecx++;
+            regs.ecx = ecx;
+            result = _CPUID(&regs);
```
Comment (anakrish) [inline tools/oesgx/oesgx.c:182]: We need to let the user know in case this _CPUID call fails.

Comment (RRathna) [reply tools/oesgx/oesgx.c:182]: The _CPUID() call in the line 159 is being reported to the user. The _CPUID() call in line 182 is to check if ecx = 0x3 (or higher) correspond to a subleaf. Is there a way to find out the number of subleafs?

Comment (anakrish) [reply tools/oesgx/oesgx.c:182]: Can the call in line 182 fail?

Comment (RRathna) [reply tools/oesgx/oesgx.c:182]: yes, and the failure is the exit condition from the while loop..

**Comment by RRathna** [comment]
Changes were tested on two devices as show below, with help from @tehsiuhuang.

daperoni@daperoni-t-n2:~/openenclave-install-2/bin$ ./oesgx
CPU supports SGX_FLC:Flexible Launch Control
CPU supports Software Guard Extensions:SGX1
MaxEnclaveSize_64: 2^(36)
CPU supports Key Sharing & Separation (KSS): false
EPC size on the platform: 58720256

acctestuser@teh-epc-yto:~/openenclave-install-2/bin$ ./oesgx
CPU supports SGX_FLC:Flexible Launch Control
CPU supports Software Guard Extensions:SGX1
MaxEnclaveSize_64: 2^(36)
CPU supports Key Sharing & Separation (KSS): false
EPC size on the platform: 117440512

**Review by tehsiuhuang — APPROVED** [review]

File: tools/oesgx/oesgx.c (hunk 1 above)
Comment (RRathna) [reply tools/oesgx/oesgx.c:191]: Added comments in src.

**Review by anakrish — APPROVED** [review]
LGTM

**Pull request merged**